tests/validation/test_input_validators.py

```diff
@@ -394,4 +394,34 @@ def test_container_registry_validator_fail_invalid_domain_format():
     with pytest.raises(ValidationError) as error:
         cv.validate(Document("example_com:443/namespace/path"))
 
-    assert str(error.value) == f"{LABEL} has invalid domain format"
+    assert (
+        str(error.value)
+        == f"{LABEL} should not include a tag, but ':443/namespace/path' was provided"
```

Comment:
I agree that by reading it a human would conclude that was likely the case, and we certainly could make some more complex rules to try to determine if a `:` is _intended_ to be a port or a tag, but intention is a hard thing to determine in general and I don't think we'll be able to fully determine which it is in all cases without quite a bit more error.

I did consider adding an extra sentence to that particular error to also check the domain, but decided against it because I thought that mistakenly including a tag would be more common than trying to include a domain that wasn't actually a domain -- but I could pretty easily be persuaded